nautobot/dcim/tests/test_models.py

```diff
@@ -1121,3 +1122,30 @@ def test_power_panel_validation(self):
             'Rack group "Rack Group 1" belongs to a location ("Location 2") that does not contain "Location 1"',
             str(cm.exception),
         )
+
+
+class InterfaceTestCase(TestCase):
+    def test_tagged_vlan_raise_error_if_mode_not_set_to_tagged(self):
+        manufacturer = Manufacturer.objects.create(name="Manufacturer 1", slug="manufacturer-1")
+        devicetype = DeviceType.objects.create(manufacturer=manufacturer, model="Device Type 1", slug="device-type-1")
+        devicerole = DeviceRole.objects.create(name="Device Role 1", slug="device-role-1")
+        site = Site.objects.create(name="Site-1", slug="site-1")
+        vlan1 = VLAN.objects.create(name="VLAN 1", vid=100, site=site)
+        status = Status.objects.get_for_model(Device)[0]
+        device = Device.objects.create(
+            name="Device 1",
+            device_type=devicetype,
+            device_role=devicerole,
+            site=site,
+            status=status,
+        )
+        interface = Interface.objects.create(
+            name="Int1",
+            type=InterfaceTypeChoices.TYPE_VIRTUAL,
+            device=device,
+        )
+        with self.assertRaises(ValidationError) as err:
+            interface.tagged_vlans.add(vlan1)
+        self.assertEqual(
+            err.exception.message_dict["tagged_vlans"][0], "Mode must be set to tagged when specifying tagged_vlans"
+        )
```

Comment:
I still feel like there should be a test here where we start with an interface with mode `tagged` and `tagged_vlans` assigned, then change it to `access` without clearing `tagged_vlans`, which should result in an error (and not the current behavior of silently clearing the `tagged_vlans`?

Reply:
Perhaps adding to the documentation that changing the interface mode from tagged to something else clears all tagged VLANs.

If that procedure could be automated, I don't believe this is required.
```json
{
   "mode": "access",
   "tagged_vlans": null,
   "untagged_vlan": "...",
}
```